### PR TITLE
Chaals more diagrams

### PR DESCRIPTION
**Pull request by chaals**
A couple more SVGs cleaned up. (98% size reduction, and the source makes sense now).

### CODE REVIEW

**Comment by chaals** [comment]
Might have a go at a couple more this week, so converting to draft.

**Comment by chaals** [comment]
working at zkp-cred-pres it occurs to me that it would be good to align the colours used with the basic credential.svg and presentation.svg. WDYT?

**Comment by chaals** [comment]
ecosystemdetail0.svg is a duplicate of ecosystemdetail.svg because there are some style changes in it that should be reviewed:
- double arrows for repeatable stuff. (I'm not convinced it's helpful...)
- dash arrays for some optional stuff
- rephrased number of times to e.g. "repeatable" / "up to once" ...
(the changes are easy to clean out, so feel free with opinions on this side of the bikeshed...).

Otherwise these are pretty much like the originals, except ~98% less code, and you can look at it for an ide a of what's going on. (And it's easy **at least for me** to make changes)

File: diagrams/credential.svg
Comment (davidlehn) [inline diagrams/credential.svg:13]: Color doesn't ~match orig.

File: diagrams/credential.svg
Comment (davidlehn) [inline diagrams/credential.svg:10]: These are not centered.  Should be x=(320-240)/2=40?

File: diagrams/presentation.svg
Comment (davidlehn) [inline diagrams/presentation.svg:9]: Probably should be centered with x=40?

File: diagrams/ecosystemdetail0.svg
Comment (davidlehn) [inline diagrams/ecosystemdetail0.svg:26]: Is the filled background needed?

Similar comment for the other diagrams with solid vs transparent backgrounds.  Should they all be the same?  And I would guess transparent?  Or is there some dark background use case where they should all be filled white?

**Comment by davidlehn** [comment]
The github image side-by-side feature is useful to look at these changes.  There are differences here and there like fonts, spacing, alignment, etc.  How close should they match?

While I do find hand optimized SVG interesting, I can't imagine people want to do that normally.  No one would start doing that vs using a tool to sketch out what things should look like.  And if some text is changed that requires boxes to be moved or re-aligned, no one wants to do math to figure that out when tools let you select some elements and center align them or similar.  What is the plan for the future of this if there is one person willing to do edits?

It is unfortunate that google docs produces large SVG.  Maybe inkscape (my preferred tool) works better?

The double lines in ecosystemdetail0.svg are confusing.  Would UMLish text like "0..*" or similar on one line be better?  Maybe no one understands that either.

File: diagrams/ecosystemdetail0.svg
Comment (dlongley) [reply diagrams/ecosystemdetail0.svg:26]: If they are transparent, they may cause trouble for people using dark mode when browsing.

File: diagrams/zkp-cred-pres.svg
Comment (brentzundel) [inline diagrams/zkp-cred-pres.svg]: all of the other examples use camel case for property names, I think this one should be as well.
"ageOver18" matches better than "Over 18 years old"

Comment (brentzundel) [inline diagrams/zkp-cred-pres.svg]: The intention for this bit of the example was not for it to be a predicate, but rather to show that the property names in the presented derived credential may not match the property names in the original credential.
so rather than "Has a Degree" I would prefer it be kept as "degree"

**Review by brentzundel — CHANGES_REQUESTED** [review]

File: diagrams/zkp-cred-pres.svg
Comment (brentzundel) [inline diagrams/zkp-cred-pres.svg]: all of the other examples use camel case for property names, I think this one should be as well.
"ageOver18" matches better than "Over 18 years old"

Comment (brentzundel) [inline diagrams/zkp-cred-pres.svg]: The intention for this bit of the example was not for it to be a predicate, but rather to show that the property names in the presented derived credential may not match the property names in the original credential.
so rather than "Has a Degree" I would prefer it be kept as "degree"

**Review by brentzundel — CHANGES_REQUESTED** [review]

File: diagrams/zkp-cred-pres.svg
Comment (brentzundel) [reply diagrams/zkp-cred-pres.svg]: something strange is happening. I'm trying to comment on line 107, but  my comment is being assigned to line 1 and immediately showing as outdated once I finish my review.

Comment (brentzundel) [reply diagrams/zkp-cred-pres.svg]: something strange is happening. I'm trying to comment on line 115, but  my comment is being assigned to line 1 and immediately showing as outdated once I finish my review.

Comment (chaals) [reply diagrams/zkp-cred-pres.svg]: somehow must be registering your selection of line in the wrong spot.

Anyway, I see the comments, and I am happy to make the change.

File: diagrams/credential.svg
Comment (chaals) [reply diagrams/credential.svg:10]: probably. I'll fix...

File: diagrams/ecosystemdetail0.svg
Comment (chaals) [reply diagrams/ecosystemdetail0.svg:26]: +1 to @dlongley - I often get caught out because don't have the same background colour as the person who created an image, and it becomes unreadable.

Suggest we resolve this comment (and make sure *all* images have a solid background...)

**Comment by chaals** [comment]
@davidlehn wrote

> The double lines in ecosystemdetail0.svg are confusing. 

I agree - let's remove the double lines.

> Would UMLish text like "0..*" or similar on one line be better? Maybe no one understands that either.

(Approximately) No one understands that either. IMHO...

File: diagrams/zkp-cred-pres.svg
Comment (chaals) [reply diagrams/zkp-cred-pres.svg:107]: (I was trying to use real text - what would be the `rdfs:label` of the property. But I'll go back on that)

Comment (chaals) [reply diagrams/zkp-cred-pres.svg]: done in 3b6f73c

Comment (chaals) [reply diagrams/zkp-cred-pres.svg]: done in 3b6f73c

File: diagrams/ecosystemdetail0.svg
Comment (chaals) [reply diagrams/ecosystemdetail0.svg:26]: background added everywhere in bd4f6b3

File: diagrams/presentation.svg
Comment (chaals) [reply diagrams/presentation.svg]: fixed in bd4f6b3

File: diagrams/credential.svg
Comment (chaals) [reply diagrams/credential.svg]: fixed in bd4f6b3

File: diagrams/credential.svg
Comment (chaals) [reply diagrams/credential.svg]: Yup. Because elsewhere, *sometimes* the claim information is presented with the same background as the proof. Should we move to consistently distinguishing claims and proof by background?

In bd4f6b3 I matched the background colour of a verifiable credential in credential.svg to the background colour used for a verifiable credential used in presentation.svg.

But I think it would be better to match the colours used in zkp-cred-pres.svg applying them to both credential.svg and presentation.svg. This should probably also apply to credential.graph and presentation.graph too.

In any event, changing colours in those files is pretty trivial to get exact matches if we want them - whether for consistency or because we decide we like some other scheme. 

I don't care what colours we use, I think it is important that they are consistent across our diagrams, and I think it is critical that we do a decent contrast check on them. (Which I have not yet done).

**Comment by chaals** [comment]
@davidlehn wrote

> It is unfortunate that google docs produces large SVG. Maybe inkscape (my preferred tool) works better?

The code from Google is not just ~50 times as large, it's also effectively impossible to make accessible without re-writing first. If Inkscape doesn't generate `text` elements for new text then it has the same problem. You should be able to test this in a minute or so by editing an image. Could you pick one and show what we get from Inkscape? (I don't have Inkscape)

I realise almost nobody wants to start by writing code. (I generally start with a pen or pencil). But most people in the group can do basic cartesian arithmetic, and adjusting the images is much less of a big deal IMHO. And if I stop responding, any reasonable SVG editing tool can work from the code I've written. If it generates garbage with each change, we're no worse off than where we started, and generally still better.

File: diagrams/claim-example-2.svg
Comment (kdenhartog) [inline diagrams/claim-example-2.svg:16]: Is this fill value correct as only 3 hex characters? The GH review is highlighting this which makes me think it needs to be 6 but I'm not an expert in SVGs.

File: diagrams/ecosystemdetail.svg
Comment (kdenhartog) [inline diagrams/ecosystemdetail.svg]: Is this one still meant to be vector based?

**Review by kdenhartog — APPROVED** [review]
I've done a manual review of this and going to tentatively approve it for now since I believe this accomplishes what we agreed to. If I get around to it I'm going to pull this branch down as well and check to make sure it all still looks fine rendered for me, but I didn't have time to do that now. If I don't get back to reviewing the rendering before someone else is ready to merge please go ahead and don't let this comment slow things done.

File: diagrams/claim-example-2.svg
Comment (chaals) [reply diagrams/claim-example-2.svg:16]: yes it's correct.

(specifically SVG converts three-character values to six by doubling up each character, so 000 -> 000000 and 4f8 -> 44ff88 etc)

File: diagrams/ecosystemdetail.svg
Comment (chaals) [reply diagrams/ecosystemdetail.svg]: this is meant to be replaced by ecosystemdetail0.svg which I'll do if there are no more comments on that.

Comment (chaals) [reply diagrams/ecosystemdetail.svg]: and done - 3f19fe8

File: diagrams/claim-example-2.svg
Comment (kdenhartog) [reply diagrams/claim-example-2.svg:16]: Ok good to know. Thanks for the insight!

**Review by brentzundel — APPROVED** [review]

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-01-12-vcwg#section4-3) on 2022-01-12

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 4.3. Chaals more diagrams (pr vc-data-model#855)

_See github pull request [vc-data-model#855](https://github.com/w3c/vc-data-model/pull/855)._




**Manu Sporny:** Set of hand-optimizations to SVG files by Charles.  
… brentz I think you were concerned about changes you requested?.  

**Brent Zundel:** I believe requested changes have been made.  

**Manu Sporny:** No objections I can see.  
… If there are no huge problems with it, we should take the PR into the change set for next week.  
… Any objections or concerns?.  
… Charles is basically optimizing and compressing SVGs..  

</details>

**Comment by kdenhartog** [comment]
I'm keen to get @davidlehn or @dlongley re-review on this in the next week before we hit the 14 day point when I merge. Notifying you now so you have a bit of time to get to it if you want any more changes before I look to merge.

**Comment by chaals** [comment]
@kdenhartog I'm happy at this point as I think it's a net improvement.

I'd still like to use colours consistently across the different diagrams at some point (i.e. all proofs use the same colour, etc) but I don't think that is so critical. And at the same time I'd like to check the contrast to make sure it's WCAG-compliant which is important (not critical given the images themselves are not critical, because the spec is well-designed).

File: diagrams/presentation-graph.svg
Comment (davidlehn) [inline diagrams/presentation-graph.svg:23]: [Unrelated to this line due to github limitiation] "AlumniOf" below should be "alumniOf".

**Review by davidlehn — CHANGES_REQUESTED** [review]
A few background and sizing issues left.

I wouldn't hold this PR up for this, but there are things that would be nice, but are challenging to do easily in hand edited images:
- center all content in the top level containers
- have consistent content border sizes in all the images
- center content within containing shapes
- align text across shapes
- align other center lines and similar
- make arrows flush on either side with source and targets
- use consistent positioning where appropriate for arrow start and end positions with source and target shapes

And also there's some style issues like claims start off with blue subject and tan claims.  But the larger pictures have same shapes and different colors.  I get why, but would be nice if colors across all images were better aligned.

File: diagrams/presentation-graph.svg
Comment (chaals) [reply diagrams/presentation-graph.svg:22]: yep.

**Comment by chaals** [comment]
I can make all the specific changes requested in comments pretty easily, whether before or after merging.

I agree with the list of inprovements in the comments (including the colour schemes, which I've commented on several times saying the same thing @davidlehn said), and while I see no reason to hold up this PR for them, I'll collect them into a new issue so I still remember if this is merged in the meantime.

**Comment by kdenhartog** [comment]
Sounds good - I'll go ahead and merge this and we can handle the additional changes with the new issue you raised.

**Comment by davidlehn** [comment]
@chaals Do we need a tracking issue for those other small changes or do you have a handle on that?

**Comment by chaals** [comment]
@davidlehn I'm good with it here. I hope I'll get to it on the weekend - I'm super-busy at the moment :S

**Pull request merged**
